src/main/java/athena/TimeAllocator.java

```diff
@@ -23,35 +25,151 @@
      * @param tasks tasks in the current TaskList
      */
     public TimeAllocator(TaskList tasks) {
-        this.flexibleTask = tasks.getFilteredList(new FlexibleTimeFilter(true)).getTasks();
-        this.fixedTask = tasks.getFilteredList(new FlexibleTimeFilter(false)).getTasks();
+        this.flexibleTaskList = tasks.getFilteredList(new FlexibleTimeFilter(true));
+        this.fixedTaskList = tasks.getFilteredList(new FlexibleTimeFilter(false));
+        this.taskList = tasks;
+
+    }
+
+    public void runAllocate() throws TaskNotFoundException {
+        LocalDate currDay = LocalDate.now();
+        ForecastFilter forecast = new ForecastFilter(Forecast.DAY);
+        for (int day = 0; day < 31; day++) {
+            System.out.println(day);
+            ArrayList<Integer> dayLog = new ArrayList<Integer>(Collections.nCopies(24, -1));
+            ArrayList<Task> predefinedTimeTasks = getSortedFixedTasks(getFixedDayTasks(currDay));
+            ArrayList<Task> undefinedTimeTasks = getSortedFixedTasks(getFlexibleDayTasks(currDay));
+
+            for (Task currTask : predefinedTimeTasks) {
+                int tag = currTask.getNumber();
+                Time timeInfo = currTask.getTimeInfo();
+                for (int i = 0; i < timeInfo.getDuration(); i++) {
+                    dayLog.set(timeInfo.getStartTime().getHour(), tag);
+                }
+            }
+
+            int start = 0;
+            int sleep = 24;
+            boolean done = false;
+            while (!done) {
+                int pos = start;
+                while (dayLog.get(pos) != -1) {
+                    pos++;
+                    if (pos == sleep) {
+                        done = true;
+                        break;
+                    }
+                }
+                int end = pos;
+                while (dayLog.get(end) == -1) {
+                    end++;
+                    if (end == sleep) {
+                        done = true;
+                        break;
+                    }
+                }
```

Comment:
It would be good to refactor this code segment into methods to maintain single level of abstraction.